### ISSUE TITLE
[feature request] custom sample size during backpropagation

### ISSUE THREAD

**Issue by tueboesen** [pytorch/pytorch#32942]
Large networks like for instance cycleGAN, will be unable to run with a batch size larger than 1-10 usually, due to the severe memory requirements during backpropagation. However this is only because the backpropagation is done for all images at once. If instead there was an optional parameter when calling backprop that set the number of images to backprop at once and then just looped through all the images then arbitrarily large batch sizes could be handled.

cc @ezyang @SsnL @albanD @zou3519 @gqchen

**Comment by albanD** [comment]
Hi,

[This](https://discuss.pytorch.org/t/why-do-we-need-to-set-the-gradients-manually-to-zero-in-pytorch/4903/20?u=alband) post on the forum details how to do this.
The important part here is that most of the memory is allocated during the forward. So the forward needs to be split as well and the forward is controlled by the user.
So we cannot make this as part of our API, the user has to do the split of the forward and use the regular API to get the gradients for what he computed.
